modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/persistence_test.ftl

```diff
@@ -330,6 +330,222 @@ public class ${entity.name}PersistenceTest {
+
+					<#if stringUtil.equals(entityColumn.type, "int")>
+						RandomTestUtil.nextInt()
+					<#elseif stringUtil.equals(entityColumn.type, "long")>
+						RandomTestUtil.nextLong()
+					<#elseif stringUtil.equals(entityColumn.type, "String")>
+						<#assign maxLength = serviceBuilder.getMaxLength(entity.getName(), entityColumn) />
+
+						<#if maxLength < 8>
+							RandomTestUtil.randomString(${maxLength})
+						<#else>
+							RandomTestUtil.randomString()
+						</#if>
+					</#if>
+
+					;
+				</#if>
+
+				${entity.name} ${entity.variableName}2 = _persistence.create(pk);
+
+				<#list entity.regularEntityColumns as entityColumn>
+					<#if !entityColumn.primary && (validator.isNull(parentPKColumn) || (parentPKColumn.name != entityColumn.name))>
+						<#if stringUtil.equals(entityColumn.type, "Blob")>
+							String ${entityColumn.name}String = RandomTestUtil.randomString();
+
+							byte[] ${entityColumn.name}Bytes = ${entityColumn.name}String.getBytes("UTF-8");
+
+							Blob ${entityColumn.name}Blob = new OutputBlob(new ByteArrayInputStream(${entityColumn.name}Bytes), ${entityColumn.name}Bytes.length);
+						</#if>
+
+						${entity.variableName}2.set${entityColumn.methodName}(
+
+						<#if stringUtil.equals(entityColumn.name, "externalReferenceCode") || stringUtil.equals(entityColumn.name, "groupId")>
+							${entity.variableName}1.get${entityColumn.methodName}()
+						<#elseif stringUtil.equals(entityColumn.name, "headId")>
+							- RandomTestUtil.nextLong()
+						<#elseif stringUtil.equals(entityColumn.type, "boolean")>
+							RandomTestUtil.randomBoolean()
+						<#elseif stringUtil.equals(entityColumn.type, "double")>
+							RandomTestUtil.nextDouble()
+						<#elseif stringUtil.equals(entityColumn.type, "int")>
+							RandomTestUtil.nextInt()
+						<#elseif stringUtil.equals(entityColumn.type, "long")>
+							RandomTestUtil.nextLong()
+						<#elseif stringUtil.equals(entityColumn.type, "BigDecimal")>
+							new BigDecimal(RandomTestUtil.nextDouble())
+						<#elseif stringUtil.equals(entityColumn.type, "Blob")>
+							${entityColumn.name}Blob
+						<#elseif stringUtil.equals(entityColumn.type, "Date")>
+							RandomTestUtil.nextDate()
+						<#elseif stringUtil.equals(entityColumn.type, "Map")>
+							new HashMap<String, Serializable>()
+						<#elseif stringUtil.equals(entityColumn.type, "String")>
+							<#assign maxLength = serviceBuilder.getMaxLength(entity.getName(), entityColumn) />
+
+							<#if maxLength < 8>
+								RandomTestUtil.randomString(${maxLength})
+							<#else>
+								RandomTestUtil.randomString()
+							</#if>
+						</#if>
+
+						);
+					</#if>
+				</#list>
+
+				_${entity.pluralVariableName}.add(_persistence.update(${entity.variableName}2));
+			}
+
+			protected ${entity.name} addDraftForExisting${entity.name}(${entity.name} existing${entity.name}) throws Exception {
+				<#if entity.hasCompoundPK()>
+					${entity.PKClassName} pk = new ${entity.PKClassName}(
```

Comment:
Are we testing this code? I think with the current service builder modules in Liferay, it is not possible to have this use case

Reply:
Yes, you're right. In fact Service Builder checks specifically that versioned entities don't have compound PKs so I will delete this part of the code.